File: project-3-test-swarm/js/test/test-weak.mjs
// WEAK suite (before the swarm): 2 checks. Run: node test/test-weak.mjs
import assert from "node:assert/strict";
import { invoice } from "../src/pricing.mjs";

const inv = invoice(10, 1);
assert.equal(inv.subtotal, 10);
assert.ok(inv.total > 0);
console.log("WEAK OK: 2 checks passed (coverage deliberately thin)");

File: project-3-test-swarm/python/tests/test_weak.py
"""WEAK suite (before the swarm): only 2 tests. Run: python3 -m unittest tests.test_weak -v"""
import os
import sys
import unittest

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "src"))

from billing import invoice


class TestWeak(unittest.TestCase):
    def test_basic_invoice(self):
        inv = invoice(10.0, 1)
        self.assertEqual(inv["subtotal"], 10.0)

    def test_total_above_subtotal_with_tax(self):
        inv = invoice(10.0, 1)
        self.assertGreater(inv["total"], 0)


if __name__ == "__main__":
    unittest.main()
